### ISSUE TITLE
calculate_number_of_cpu validates its argument with assert

### ISSUE THREAD

**Issue by bendichter** [NeurodataWithoutBorders/nwbinspector#755]
`calculate_number_of_cpu` in `utils/_utils.py` validates its argument with `assert` statements:

```python
assert requested_cpu <= total_cpu, f"Requested more CPUs ({requested_cpu}) than are available ({total_cpu})!"
assert requested_cpu >= -(total_cpu - 1), f"Requested fewer CPUs ({requested_cpu}) than are available ({total_cpu})!"
```

`assert` is stripped when Python runs with `-O`, so under optimized bytecode an out-of-range `--n-jobs` value is passed through to `ProcessPoolExecutor` unchecked. Argument validation in library code should raise `ValueError`. The second message is also misleading, since it fires when the requested value is too negative rather than when fewer CPUs were requested than exist.

Since callers may currently catch `AssertionError`, this is a small behavior change and the tests that check for it need to be updated.
